rubicon/objc/objc.py

```diff
@@ -17,16 +18,34 @@
 
 ######################################################################
 
-def _find_or_error(name):
+_lib_path = ["/usr/lib"]
+_framework_path = ["/System/Library/Frameworks"]
+def _load_or_error(name):
     path = util.find_library(name)
-    if path is None:
-        raise ValueError("Library {!r} not found".format(name))
-    else:
-        return path
+    if path is not None:
+        return CDLL(path)
+
+    # On iOS (and probably also watchOS and tvOS), ctypes.util.find_library doesn't work and always returns None.
+    # This is because the sandbox hides all system libraries from the filesystem and pretends they don't exist.
+    # However they can still be loaded if the path is known, so we try to load the library from a few known locations.
```

Comment:
No, I broke it, sort of. Here's what happened: `ctypes.util.find_library` returns `None` if the requested library is not found. And if you pass `None` as the library name to `ctypes.CDLL`, it means that rather than loading a specific library, you get a `CDLL` that gives access to all currently loaded global functions. That means that previously when `ctypes.util.find_library` returned `None` on iOS, it silently switched to "ALL the functions" mode instead of raising an error.

In 53b0c4bbc9b110208043a8f714fd115356b55d8b I added a `_find_or_error` utility function that raised an error when the library was not found, instead of silently returning `None`, so I would notice if I mistyped a library name. On macOS this worked fine, because `ctypes.util.find_library` works correctly there. On iOS it broke, because now the "not found" case was treated as a hard error, and no longer passed the `None` on to `CDLL`.